src/data/fin.lean

```diff
@@ -1177,6 +1198,23 @@ begin
       exact lt_succ_iff.mpr (not_lt.mp h), }, },
 end
 
+/-- `pred` commutes with `succ_above`. -/
+lemma pred_succ_above_pred {a : fin (n + 2)} {b : fin (n + 1)} (ha : a ≠ 0) (hb : b ≠ 0)
+  (hk := succ_above_ne_zero ha hb) :
+  (a.pred ha).succ_above (b.pred hb) = (a.succ_above b).pred hk :=
+begin
+  have hbc : (b.pred hb).cast_succ = b.cast_succ.pred _, by { cases b, refl },
```

Comment:
Should this be a lemma by itself?

Reply:
this again uses an implicit proof parameter (this is what the trailing `simpa` deals with), would that be a good idea?

Reply:
Yes, I think it would be a good idea

Reply:
done in c554867